File: 2016/d08_test.rb
# frozen_string_literal: true

require 'minitest/autorun'
require_relative 'd08'

class TestD08 < Minitest::Test
  def setup
    @example = <<~EXAMPLE
      rect 3x2
      rotate column x=1 by 1
      rotate row y=0 by 4
      rotate column x=1 by 1
    EXAMPLE
  end

  def test_part1
    assert_equal 6, D08.new(@example).part1
  end
end

File: 2018/test/aoc2018/d08_test.clj
(ns aoc2018.d08-test
  (:require [clojure.test :refer [deftest is]]
            [aoc2018.d08 :as d]))

(def example "2 3 0 3 10 11 12 1 1 0 1 99 2 1 1 2")

(deftest part1-test
  (is (= 138 (d/part1 example))))

(deftest part2-test
  (is (= 66 (d/part2 example))))
